lib/SampleService/core/config.py

```diff
@@ -198,24 +239,89 @@ def _check_string_req(s: Optional[str], name: str) -> str:
     :returns: A set of metadata validators.
     '''
     # TODO VALIDATOR make validator CLI
+
     try:
-        with _urllib.request.urlopen(url) as res:
-            cfg = _yaml.safe_load(res)
+        config_asset = None
+        if url:
+            config_url = url
+        elif not repo_path:
+            raise ValueError(f'No metadata validator config URL or repo path.')
+        else:
+            if not repo_asset:
+                raise ValueError(f'No repo_asset name provided for repo "{repo_path}"')
+            try: 
+                repo = _Github(login_or_token=token).get_repo(repo_path)
+                releases = [rel for rel in repo.get_releases() if prerelease_ok or not rel.prerelease]
+            except Exception as e:
+                raise RuntimeError(f'Fetching releases from repo "{repo_path}" failed.') from e
+
+            if not releases:
+                raise ValueError(f'No releases found in validator config repo "{repo_path}"')
+
+            if tag:
+                target_release = next((r for r in releases if r.tag_name==tag), None)
+                if not target_release:
+                    raise ValueError(f'No release with tag "{tag}" found in validator config repo "{repo_path}"')
+            else:
+                # Use the latest release
+                target_release = releases[0]
+
+            assets = target_release.get_assets()
+            if not assets:
+                raise ValueError(f'No assets found in validator config repo "{repo_path}", '+
+                    f'release tag "{target_release.tag_name}"')
+
+            config_asset = next((a for a in assets if a.name==repo_asset), None)
+            if not config_asset:
+                raise ValueError(f'No config asset "{repo_asset}" found in validator config '+
+                    f'repo "{repo_path}", release tag "{target_release.tag_name}"')
+
+            config_url = config_asset.url
+
+        req = _request.Request(config_url)
+        req.add_header('Accept', 'application/octet-stream')
+        if token:
+            req.add_header('Authorization', f'token {token}')
+        with _request.urlopen(req) as response:
+            cfg = _yaml.safe_load(response)
+
     except _URLError as e:
-        raise ValueError(
-            f'Failed to open validator configuration file at {url}: {str(e.reason)}') from e
+        if config_asset:
+            raise ValueError(f'Error downloading config asset from repo "{repo_path}" '+
+                f'tag "{tag}" asset "{config_asset.url}": {str(e.reason)}') from e
+        else:
+            raise ValueError(f'Error downloading config asset from {url}: {str(e.reason)}') from e
     except _ParserError as e:
-        raise ValueError(
-            f'Failed to open validator configuration file at {url}: {str(e)}') from e
+        if config_asset:
+            raise ValueError(
+                f'Failed to open validator configuration file from repo "{repo_path}" '+
```

Comment:
Mentioned I think in a previous comment, I'm not of the opinion one is needed, as this is the same (already tested) bad-yaml case with different err text. Covered by test_config_get_validators_fail_bad_yaml

Reply:
I guess I don't understand why you wouldn't add a test? It's an uncovered code path and so should get a test for that code path IMO. The tests look pretty simple to write so naively it seems like a 5 minutes thing. I would certainly write one if it was my PR

Reply:
Mostly because it's a weird amount of overhead to use a third party service in tests to test a code path that is only differentiated by earlier (tested) behavior. It's also not a functionally different code path, it just changes what string is thrown...